tests/test_ada_url.py

```diff
@@ -56,6 +83,24 @@ def test_class_set(self):
         expected = 'wss://changed-host:9090/new-pathname?new-search#new-hash'
         self.assertEqual(actual, expected)
 
+    def test_class_delete(self):
+        url = 'https://user_1:password_1@example.org:8080/dir/../api?q=1#frag'
+        urlobj = URL(url)
+
+        del urlobj.port
+        self.assertEqual(
+            urlobj.href, 'https://user_1:password_1@example.org/api?q=1#frag'
+        )
+
+        del urlobj.hash
+        self.assertEqual(urlobj.href, 'https://user_1:password_1@example.org/api?q=1')
+
+        del urlobj.pathname
+        self.assertEqual(urlobj.href, 'https://user_1:password_1@example.org?q=1')
```

Comment:
@anonrig - this looked weird to me, because I think this is not allowed by [RFC1738](https://www.rfc-editor.org/rfc/rfc1738).

It does pass `ada_check_url`, so it's not just an artifact of deleting. Is it right, or is `ada_url` wrong? I think I'm missing where the WHATWG spec allows this.

Reply:
>  this looked weird to me


Please refer to the WHATWG URL specification: 
<img width="1135" alt="Capture d’écran, le 2023-08-28 à 17 26 45" src="https://github.com/ada-url/ada-python/assets/391987/6da190da-09e9-4aac-ac72-f8b143f91876">

Once you enter the 'path' state, if '?' is encountered, then you move directly to the query processing.

https://url.spec.whatwg.org

Reply:
[Here](https://jsdom.github.io/whatwg-url/#url=aHR0cHM6Ly91c2VyXzE6cGFzc3dvcmRfMUBleGFtcGxlLm9yZz9xPTE=&base=YWJvdXQ6Ymxhbms=) is the `jsdom` live URL viewer for this input. Note that it changes the `href` to include the `/`:

![image](https://github.com/ada-url/ada-python/assets/1922815/febd5e58-a0e3-4c1b-ab7f-99313819edfd)

So I think either `ada_url` or `jsdom/whatwg-url` is wrong about what the `href` should be, yes?

Reply:
Does `del url.pathname` trigger `ada_clear_pathname` function? If yes, then we have a bug. Because if the URL has a hash, the pathname has to be `/` - even if it's deleted. If that's the case, can you open an issue to `ada` repository? I'll fix it as soon as possible.

Reply:
Yes, it calls the clear method. I'll open an issue. Thanks!

Reply:
https://github.com/ada-url/ada/issues/490 is the issue. I'll hold this release for that fix, since the Python bindings don't expose the `clear` methods yet.